src/simulation/m_data_output.fpp

```diff
@@ -1748,35 +1794,61 @@ contains
         !!      other procedures that are necessary to setup the module.
     impure subroutine s_initialize_data_output_module
 
+        integer :: i, m_ds, n_ds, p_ds
+
         ! Allocating/initializing ICFL, VCFL, CCFL and Rc stability criteria
-        @:ALLOCATE(icfl_sf(0:m, 0:n, 0:p))
-        icfl_max = 0._wp
+        if (run_time_info) then
+            @:ALLOCATE(icfl_sf(0:m, 0:n, 0:p))
+            icfl_max = 0._wp
+
+            if (viscous) then
+                @:ALLOCATE(vcfl_sf(0:m, 0:n, 0:p))
+                @:ALLOCATE(Rc_sf  (0:m, 0:n, 0:p))
+
+                vcfl_max = 0._wp
+                Rc_min = 1.e3_wp
+            end if
+        end if
 
         if (probe_wrt) then
             @:ALLOCATE(c_mass(num_fluids,5))
         end if
 
-        if (viscous) then
-            @:ALLOCATE(vcfl_sf(0:m, 0:n, 0:p))
-            @:ALLOCATE(Rc_sf  (0:m, 0:n, 0:p))
+        if (down_sample) then
```

Comment:
shouldn't the routines that do the up/down sampling go into common? so you can use them in pre/post/sim as needed without redefining variables.

Reply:
I'm not sure which routines you're referring to. I have downsampling in common and upsampling in sim_helpers since it's only used in simulation

Reply:
Moved upsample to `common/m_sim_helpers`

Reply:
yeah got it